tests/roles/nova_adoption/tasks/main.yaml

```diff
@@ -0,0 +1,100 @@
+              [workarounds]
+              disable_compute_service_check_for_ffu=true
+          metadataServiceTemplate:
+            enabled: true # deploy single nova metadata on the top level
+            override:
+              service:
+                metadata:
+                  annotations:
+                    metallb.universe.tf/address-pool: internalapi
+                    metallb.universe.tf/allow-shared-ip: internalapi
+                    metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+                spec:
+                  type: LoadBalancer
+            customServiceConfig: |
+              [workarounds]
+              disable_compute_service_check_for_ffu=true
+          schedulerServiceTemplate:
+            customServiceConfig: |
+              [workarounds]
+              disable_compute_service_check_for_ffu=true
+          cellTemplates:
+            cell0:
+              conductorServiceTemplate:
+                customServiceConfig: |
+                  [workarounds]
+                  disable_compute_service_check_for_ffu=true
+              metadataServiceTemplate:
+                enabled: false # enable here to run it in a cell instead
+                override:
+                    service:
+                      metadata:
+                        annotations:
+                          metallb.universe.tf/address-pool: internalapi
+                          metallb.universe.tf/allow-shared-ip: internalapi
+                          metallb.universe.tf/loadBalancerIPs: 172.17.0.80
+                      spec:
+                        type: LoadBalancer
+                customServiceConfig: |
+                  [workarounds]
+                  disable_compute_service_check_for_ffu=true
+            cell1:
+              conductorServiceTemplate:
+                customServiceConfig: |
+                  [workarounds]
+                  disable_compute_service_check_for_ffu=true
+    '
+
+- name: wait for Nova to start up
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc get pod --selector=component=nova-scheduler -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+    oc get pod --selector=component=nova-api -o jsonpath='{.items[0].status.phase}{"\n"}' | grep Running
+  register: nova_running_result
+  until: nova_running_result is success
+  retries: 180
+  delay: 2
+
+- name: check that Nova is reachable and its endpoints are defined
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    alias openstack="oc exec -t openstackclient -- openstack"
+
+    ${BASH_ALIASES[openstack]} endpoint list | grep nova
+    ${BASH_ALIASES[openstack]} server list
+  register: nova_responding_result
+  until: nova_responding_result is success
+  retries: 60
+  delay: 2
+
+- name: Nova online data migrations
+  ansible.builtin.shell: |
+    {{ shell_header }}
+    {{ oc_header }}
+    oc exec -it nova-scheduler-0 -- nova-manage db online_data_migrations
```

Comment:
@gibizer @SeanMooney I took this from cinder tasks, but perhaps we need somthing similar for Nova post-adoption?

Reply:
it seems even if we need any migrations applied, we cannot do this early, but in another role, say ``nova_adoption_cleanup``, after dataplane adoption upgrades and adopts nova computes services

Reply:
this should not be done in ansibel really at all even in another role

also if we needed to do this we would have to do it via the conductor contiaen. the scheduled wont have access to all the relevant dbs.

this really should be done by the nova operator isntead.

Reply:
@jistr @gibi @slagle @bshephar ^^ the above makes sense, although how could the nova operator know when EDPM adoption ansible tasks completes so that it can run migrations? And then signal it back to unpatch the FFU config workarounds?

tbh, this looks overengineered. We must use ansible for all that

Reply:
@jistr @gibi @slagle @bshephar ^^ the above makes sense. Although, how could the nova operator know **when** EDPM adoption ansible tasks completes, so that it could run migrations afterwards? And **then** signal it (adoption tasks) back to unpatch the FFU config workarounds?

tbh, this looks overengineered. We must use ansible for all that

Reply:
Generally i'd vote for just putting together the procedure in bash even if it's complex (unless it's extremely brutally complex :) ), and then we could discuss if we need automation (Ansible/bash scripts). We've been trying to stay away from "shipping Adoption as code" if possible. If that becomes unbearable in some places, we would reconsider.  (And it's one of the next gen mtg topics.)

Reply:
If we need to do this after the "dataplane_adoption" doc/role, then we can either put it there, or better yet i would be +1 on creating an additional doc (+ role in the tests), something like "finalize_adoption" or "post_adoption". That would go at the end of the list, and multiple services could put snippets there that need to run at the very end of the Adoption procedure, like the Nova online migration IIUC.

Reply:
:+1: Thanks